File: bubble_sort.js
Array.prototype.bubbleSort = function(){
  var array = this
  var size = array.length
  var tmp
  for(var i = 0; i < size; i++)
    for(var j = 0; j < size; j++)
      if(array[i] > array[j]){
        tmp = array[j]
        array[j] = array[i]
        array[i] = tmp
      }
}

File: bubble_sort.rb
class Array

  def bubble_sort!
    tap do |array|
      size = length
      size.times do |i|
        size.times do |j|
          if array[i] > array[j]
            array[i], array[j] = array[j], array[i]
          end
        end
      end
    end
  end

  def fair_bubble_sort!
    tap do |array|
      size = length
      tmp = nil
      i = 0
      while i < size
        j = 0
        while j < size
          if array[i] > array[j]
            tmp = array[i]
            array[i] = array[j]
            array[j] = tmp
          end
          j += 1
        end
        i += 1
      end
    end
  end

end
